HIP/hip-583.md

```diff
@@ -10,220 +10,451 @@ status: Accepted
-Note, in this case the final value of the account alias will remain as the `public-address` and not the `public-key` as it would be in the `auto-create` case.
+If the `alias` is used during `CryptoTransfer` as the _target_ of the transfer, then there is no need to prove that
+anyone controls that account. If an `alias` is used in the `AccountID` of the payer, then ownership must be asserted,
+just as is done for any other transaction today.
 
-As an illustration, a wallet or exchange could create and fund a new Hedera account based on the `public-address` alone. Once the transaction goes through the owner of the `public-address` user may use a wallet of their choice to submit a signed transaction (e.g. `CryptoTransfer`, `ContractCreate`, `ContractCall`, `EthereumTransaction` or `TokenAssociation`) to claim ownership of the account and carry out the desired transaction. The ownership claim and account completion will be seamless and transparent to the user.
+#### Validation of the `alias`
 
-`CryptoTransfer` transactions should also support this case.
-This will support users, developers and exchanges who prefer to create and credit an accounts balance in one step.
+It **is** possible to specify an `account-num alias` as the _target_ of a `CryptoTransfer` transaction. Indeed, for
+Ethereum-native wallets, this is the only way to transfer value to Hedera accounts that don't have an
+`evm address alias`. However, it is **impossible** to auto-create an account using the `account-num alias`. If the
+account does not exist, the transfer will fail.
 
-`EthereumTransaction` transactions where value is transfered  (i.e `tx.to` and `tx.value` are set) should also support this case.
-This will support users of `Ethereum-native wallets` and eth dev tools who are accumstomed to creating and crediting an accounts balance in one step.
-In this case th `tx.to` value of the RLP encoded transaction would dictate the `public-address` to be used as an alias.
+##### `key alias`
 
-### Allow CryptoCreate to automatically set alias
+The `key alias` is a protobuf-encoded `Key` object. We only support `ED25519` and `ECDSA_secp256k1` keys at this time.
+When an attempt is made to create a new account with `CryptoCreate` with an alias, we check whether it is a `key alias`
+by checking if there are more than 20 bytes. We then:
 
-When `CryptoCreate.key` is an `ECDSA` key, the network node may calculate the `public-address` and automatically set the accounts alias value. 
+1. Deserialize the protobuf object. If the object is not a protobuf object, we reject the transaction.
+2. If the protobuf object is not an `ED25519` or `ECDSA_secp256k1` `Key`, we reject the transaction.
+3. If the `Key` is an `ED25519` key:
+    1. If the alias is already in use, we use that account and do not create a new one
```

Comment:
We would actually reject the transaction since this is a `CryptoCreate` and not a `CryptoTransfer`

Reply:
Correct! Fixing the doc.

Reply:
Ah, I see the problem. This section is talking about CryptoTransfer, but mistakingly says it is CryptoCreate. I will correct.

Reply:
Done.